### ISSUE TITLE
Media plugin media_url_resolver issue

### ISSUE THREAD

**Issue by axww** [tinymce/tinymce#8241]
Hi,
In Media plugin, the media_url_resolver option has problem.
While the function changes Media window embeded code (html/iframe), the url did not change.
If I change width/height options, the e cmbeded ode will change to iframe with origin url (Not the function changed one).
Please check if you can fix it, thank you!

**Comment by lostkeys** [comment]
Hi @aaa-aa-a. Sorry it's a bit hard to follow exactly wha the issue is. Can you clarify and provide a clear reproduction case? Preferable in a [TinyMCE Fiddle](https://fiddle.tiny.cloud).

**Comment by aaa-aa-a** [comment]
@lostkeys 
Hi, thank you for following the issue!

Here's my sample code on https://fiddle.tiny.cloud/
```
<script type="text/javascript">
  tinymce.init({
    selector: "textarea",
    plugins: ["media"],
    toolbar: "media",
    media_url_resolver: (data, resolve/*, reject*/) => {
      if (data.url.indexOf('a.com') !== -1) {
        resolve({ html: `<iframe src="https://z.com/" width="400" height="400" ></iframe>` });
      } else {
        resolve({ html: '' });
      }
    }
  });
</script>
<form method="post" action="dump.php"><textarea name="content"></textarea></form>
```

Click "Media" button then input "a.com" in "Source" box, the code in "Embed" will become:
`<iframe src="https://z.com/" width="400" height="400" ></iframe>`
That's correct, the situation using regular expression to match and change into iframe with different URL.

However, in "General" tab, if I change any of "width" and "height" to "500", the code in "Embed" will become:
`<iframe src="a.com" width="500" height="500"></iframe>`
That's wrong, the result I wish to see is the "src" in <iframe> remain `"https://z.com/"`, however it changes to the "url" value in "General" tab.

The issue here is when width/height changes in "General", the "Embed" iframe code will automatically re-format using parameters in "General", but media_url_resolver did not active when parameters changes.

Thanks for your kind help!
Best regards

**Comment by axww** [comment]
TinyMCE 7 using Promise, issue still exists.

**Comment by axww** [comment]
Same issue here:
https://github.com/tinymce/tinymce/issues/9798
